zerver/tests/test_subs.py

```diff
@@ -1220,78 +1323,113 @@ def test_cant_remove_others_from_stream(self) -> None:
+    def test_realm_admin_remove_multiple_users_from_stream(self) -> None:
         """
-        If you're an admin, you can remove multiple users from a stream,
+        If you're a realm admin, you can remove multiple users from a stream,
         """
         result = self.attempt_unsubscribe_of_principal(
-            query_count=30, target_users=[self.example_user('cordelia'), self.example_user('prospero')],
-            is_admin=True, is_subbed=True, invite_only=False, target_users_subbed=True)
+            query_count=32, target_users=[self.example_user('cordelia'), self.example_user('prospero')],
+            is_realm_admin=True, is_subbed=True, invite_only=False, target_users_subbed=True)
         json = self.assert_json_success(result)
         self.assertEqual(len(json["removed"]), 2)
         self.assertEqual(len(json["not_removed"]), 0)
 
-    def test_admin_remove_others_from_subbed_private_stream(self) -> None:
+    def test_realm_admin_remove_others_from_subbed_private_stream(self) -> None:
         """
-        If you're an admin, you can remove other people from private streams you
+        If you're a realm admin, you can remove other people from private streams you
         are on.
         """
         result = self.attempt_unsubscribe_of_principal(
-            query_count=21, target_users=[self.example_user('cordelia')], is_admin=True, is_subbed=True,
-            invite_only=True, target_users_subbed=True)
+            query_count=23, target_users=[self.example_user('cordelia')], is_realm_admin=True,
+            is_subbed=True, invite_only=True, target_users_subbed=True)
         json = self.assert_json_success(result)
         self.assertEqual(len(json["removed"]), 1)
         self.assertEqual(len(json["not_removed"]), 0)
 
-    def test_admin_remove_others_from_unsubbed_private_stream(self) -> None:
+    def test_realm_admin_remove_others_from_unsubbed_private_stream(self) -> None:
         """
-        If you're an admin, you can remove people from private
+        If you're a realm admin, you can remove people from private
         streams you aren't on.
         """
         result = self.attempt_unsubscribe_of_principal(
-            query_count=21, target_users=[self.example_user('cordelia')], is_admin=True, is_subbed=False,
-            invite_only=True, target_users_subbed=True, other_sub_users=[self.example_user("othello")])
+            query_count=22, target_users=[self.example_user('cordelia')], is_realm_admin=True,
+            is_subbed=False, invite_only=True, target_users_subbed=True, other_sub_users=[self.example_user("othello")])
+        json = self.assert_json_success(result)
+        self.assertEqual(len(json["removed"]), 1)
+        self.assertEqual(len(json["not_removed"]), 0)
+
+    def test_stream_admin_remove_others_from_public_stream(self) -> None:
+        """
+        If you're a stream admin, you can remove people from public streams.
```

Comment:
Reword this; should say "You can remove people from public streams you're a stream administrator for."

And same for the others here.  Critically, the behavior is only for the streams you administer, and we should make sure these comments don't mislead on what we're testing.